src/DTS/eBaySDK/Trading/Types/AbstractRequestType.php

```diff
@@ -48,7 +48,7 @@ class AbstractRequestType extends \DTS\eBaySDK\Types\BaseType
             'elementName' => 'RequesterCredentials'
         ),
         'DetailLevel' => array(
-            'type' => 'string',
+            'type' => '\DTS\eBaySDK\Trading\Enums\DetailLevelCodeType[]',
```

Comment:
I can see what you are trying to do here but it does not work. Both the following lines will throw an exception because of this change.

```php
$request->DetailLevel = 'ReturnAll';

$request->DetailLevel = Enums\DetailLevelCodeType::C_RETURN_ALL;
```

In addition I'm not sure if the SDK will be able to check that a constant has been passed to it. All the SDK will see is that a string value has been assigned.